script/utils/batch-diff-etherscan.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# chain ID is hardcoded
+CHAIN_ID=8453 # base
+# CHAIN_ID=84532 # base-sep
+
+echo "=== Batch Contract Diff ==="
+echo "Using Chain ID: $CHAIN_ID (change in script as needed)"
+echo "1. Paste your contract table below (copied from raw or rendered markdown). ContractName assumed to be last column."
+echo "2. After pasting, press Enter"
+echo "3. Then press Ctrl+D to finish input :"
+# Create a temporary file to store the pasted table
+TEMP_FILE=$(mktemp)
+cat > "$TEMP_FILE"
+
+# Initialize counter for commands run
+COMMANDS_RUN=0
+
+# Process each line of the file
+while IFS= read -r line; do
+    # Only process lines that contain /address/
+    if [[ "$line" == *"/address/"* ]]; then
+        # Extract the address using regex
+        if [[ "$line" =~ (0x[a-fA-F0-9]{40}) ]]; then
+            ADDRESS="${BASH_REMATCH[1]}"
```

Comment:
went to run this on my mac and since it auto uses zsh this failed. I can use bash if I have to up would prefer updating the script to something that works in both.

Reply:
@kutoft feel free to add commits here for an alternative that works for mac too (I can't test it)